realsense_camera/src/base_nodelet.cpp

```diff
@@ -182,15 +150,13 @@ namespace realsense_camera
     }
 
     // Print list of all cameras found
-    listCameras(num_of_cameras);
-
-    pnh_.getParam("serial_no", serial_no);
-    pnh_.getParam("usb_port_id", usb_port_id);
+    int type_count = listCameras(num_of_cameras);
```

Comment:
There needs to be a check here if type_count is > 0
`  // Exit with error if no cameras of correct type are connected.`
`    if (type_count < 1)`
`   {`
`      ROS_ERROR_STREAM(nodelet_name_ << " - No '" << camera_type_ << "' cameras detected!");`
`      rs_delete_context(rs_context_, &rs_error_);`
`      checkError();`
`      return false;`
`    }`